### ISSUE TITLE
Error connecting to TLS 1.3 Only Domain

### ISSUE THREAD

**Issue by rjhancock** [getinsomnia/insomnia#6948]
### Expected Behavior

Connection to occur with a proper handshake and to receive "Abandon Hope All Who Come Here" as the body of the response when making a GET request to `https://api.fasa.dev`

### Actual Behavior

Receive the following in the Timeline panel of Insomnia 8.5.1

```
* Preparing request to https://api.fasa.dev/
* Current time is 2023-12-24T02:00:57.967Z
* Enable automatic URL encoding
* Using default HTTP version
* Enable timeout of 30000ms
* Enable SSL validation
* Enable cookie sending with jar of 1 cookie
* Too old connection (392 seconds), disconnect it
* Connection 9 seems to be dead!
* Closing connection 9
*   Trying 157.230.8.49:443...
* Connected to api.fasa.dev (157.230.8.49) port 443 (#11)
* ALPN, offering h2
* ALPN, offering http/1.1
* A bad protocol version was encountered
* Closing connection 11
```

### Reproduction Steps

Create a new `GET` route and set it to `https://api.fasa.dev` and click `Send`

### Is there an existing issue for this?

- [X] I have searched the [issue tracker](https://www.github.com/Kong/insomnia/issues) for this problem.

### Additional Information

When using `curl -v https://api.fasa.dev` I get the following block of text as well as the correct response within a browser.

```
*   Trying 2604:a880:400:d0::1d81:f001:443...
* Connected to api.fasa.dev (2604:a880:400:d0::1d81:f001) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* TLSv1.0 (OUT), TLS header, Certificate Status (22):
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS header, Certificate Status (22):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS header, Finished (20):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.2 (OUT), TLS header, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: CN=fasa.dev
*  start date: Dec 19 21:07:10 2023 GMT
*  expire date: Mar 18 21:07:09 2024 GMT
*  subjectAltName: host "api.fasa.dev" matched cert's "*.fasa.dev"
*  issuer: C=US; O=Let's Encrypt; CN=R3
*  SSL certificate verify ok.
* Using HTTP2, server supports multiplexing
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* Using Stream ID: 1 (easy handle 0x55b241e75e90)
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
> GET / HTTP/2
> Host: api.fasa.dev
> user-agent: curl/7.81.0
> accept: */*
> 
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* old SSL session ID is stale, removing
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* Connection state changed (MAX_CONCURRENT_STREAMS == 128)!
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
< HTTP/2 200 
< server: nginx
< date: Sun, 24 Dec 2023 02:05:15 GMT
< content-type: text/plain; charset=utf-8
< content-length: 30
< vary: Accept-Encoding
< x-xss-protection: 1; mode=block
< x-content-type-options: nosniff
< referrer-policy: no-referrer-when-downgrade
< permissions-policy: interest-cohort=()
< strict-transport-security: max-age=31536000; includeSubDomains; preload
< 
* Connection #0 to host api.fasa.dev left intact
Abandon Hope All Who Come Here
```

Nginx is configured for SSL with the following protocols:

`    ssl_protocols          TLSv1.3;`

### Insomnia Version

8.5.1

### What operating system are you using?

macOS

### Operating System Version

Sonoma 14.3

### Installation method

Downloaded from Site.

**Comment by subnetmarco** [comment]
Hello, this seems to be a regression and we are actively working to fix it.

**Comment by rjhancock** [comment]
Thought it was odd as it was working before. Thank you.

**Comment by notjaywu** [comment]
@rjhancock [Insomnia 8.5.2-beta.0 📦](https://github.com/Kong/insomnia/releases/tag/core%408.5.2-beta.0) is out to fix this issue, please take a try, thank you.

**Comment by rjhancock** [comment]
A quick test shows fix confirmed.

**Comment by rjhancock** [comment]
I looked at the related ticket for switching to OpenSSL and looked at the Apple Docs, looks like moving from SecureTransport to the Network framework might have solved the issue as well and allowed you to keep access to Keychain.

Would that work to keep UX as previous based upon the ticket?

**Comment by notjaywu** [comment]
It has now been fixed in 8.6.0+.
